src/main/kotlin/solve/importer/view/ProjectTreeView.kt

```diff
@@ -0,0 +1,97 @@
+import solve.utils.loadImage
+import tornadofx.*
+
+open class ProjectTreeView : View() {
+
+    private var rootTree = TreeItem(FileInTree(FileInfo("", false)))
+
+    private val directoryPathView: DirectoryPathView by inject()
+
+    private fun updateTree() {
+        directoryPathView.path.onChange {
+            if (!it.isNullOrEmpty()) {
+                rootTree.children.remove(0, rootTree.children.size)
+                if (directoryPathView.project.value != null) {
+                    rootTree = ProjectParser.createTreeWithFiles(directoryPathView.project.value, rootTree)
+                }
+            }
+        }
+    }
+
+    override val root = treetableview(rootTree) {
+        visibleWhen { directoryPathView.project.isNotNull }
+        this.isShowRoot = false
+
+        val filesColumn: TreeTableColumn<FileInTree, FileInfo> = TreeTableColumn<FileInTree, FileInfo>().apply {
+            isResizable = false
+            prefWidth = 220.0
+        }
+        val errorsColumn: TreeTableColumn<FileInTree, FileInfo> = TreeTableColumn<FileInTree, FileInfo>().apply {
+            isResizable = false
+            prefWidth = 150.0
+        }
+
+        filesColumn.cellValueFactory = TreeItemPropertyValueFactory("file")
+        errorsColumn.cellValueFactory = TreeItemPropertyValueFactory("file")
+
+        filesColumn.setCellFactory { _ ->
+            object : TreeTableCell<FileInTree, FileInfo?>() {
+                private val imageIcon = loadImage("icons/importer/photo.png")
+                private val fileIcon = loadImage("icons/importer/description.png")
+                private val errorFolderIcon = loadImage("icons/importer/error_folder.png")
+                private val errorFileIcon = loadImage("icons/importer/error_file.png")
+
+                override fun updateItem(item: FileInfo?, empty: Boolean) {
+                    super.updateItem(item, empty)
+                    text = if (empty) null else item!!.name
+                    graphic = if (empty) {
+                        null
+                    } else {
+                        if (item!!.isLeaf) {
+                            if (item.error.isEmpty()) {
+                                ImageView(fileIcon)
+                            } else {
+                                ImageView(errorFileIcon)
+                            }
+                        } else {
+                            if (item.error.isEmpty()) {
+                                ImageView(imageIcon)
+                            } else {
+                                ImageView(errorFolderIcon)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        errorsColumn.setCellFactory {
+            object : TreeTableCell<FileInTree, FileInfo?>() {
+                override fun updateItem(item: FileInfo?, empty: Boolean) {
+                    super.updateItem(item, empty)
+                    if (empty) {
```

Comment:
use ternary conditional

Reply:
there is no ternary operator(a ? b : c)  in Kotlin